internal/internal_workflow_client.go

```diff
@@ -260,6 +260,40 @@ func (wc *workflowClient) ExecuteWorkflow(ctx context.Context, options StartWork
 	}, nil
 }
 
+// GetWorkflow gets a workflow execution and returns a WorkflowRun that will allow you to wait until this workflow
+// reaches the end state, such as workflow finished successfully or timeout.
+// The current timeout resolution implementation is in seconds and uses math.Ceil(d.Seconds()) as the duration. But is
+// subjected to change in the future.
+func (wc *workflowClient) GetWorkflow(ctx context.Context, workflowID string, runID string) (WorkflowRun, error) {
+
+	// retrieve the workflow execution information
+	resp, err := wc.DescribeWorkflowExecution(ctx, workflowID, runID)
```

Comment:
I think we might only need to call DescribeWorkflowExecution when runID is empty, so that user will be able to get not only the current run.

Reply:
I think there is no need to call it at all as GetWorkflowHistory should work with empty runID.

Reply:
What is the correct way to check for existence of workflow? I could just pass back a WorkflowRun object based of the workflowID and runID passed in, but after some validation perferable.. .